javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/TracerInstaller.java

```diff
@@ -99,6 +108,23 @@ private static MetricExporterFactory findMetricExporterFactory(String exporterNa
     return null;
   }
 
+  private static MetricProducerCustomizer findMetricCollector(String exporterName) {
+    ServiceLoader<MetricProducerCustomizer> serviceLoader =
+        ServiceLoader.load(MetricProducerCustomizer.class, TracerInstaller.class.getClassLoader());
+
+    for (MetricProducerCustomizer metricProducerCustomizer : serviceLoader) {
+      if (metricProducerCustomizer
+          .getClass()
+          .getSimpleName()
+          .replace("_", "")
+          .toLowerCase()
+          .startsWith(exporterName.toLowerCase())) {
```

Comment:
Instead of this undocumented naming convention what do you think about adding a `String getExporterName()` method to the `MetricProducerCustomizer` interface?

Reply:
ya, this has come up for `SpanExporterFactory` and `MetricExporterFactory` also https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1253#discussion_r494221024 😄

I will create a tracking issue for this

Reply:
#1287

Reply:
@iNikem the metrics can use the pull and push model. The p8s metric exporter uses pull model. It pulls metrics from OTEL once the metric endpoint is hit.

Reply:
But push model is supported by our Exporters API, they don't need any new SPI, right?

Reply:
@iNikem `MetricExporter` supports push model, but not pull model. Pull model has to use `MetricProducer` directly, e.g. https://github.com/open-telemetry/opentelemetry-java/blob/a08dd9fbe4a98f34aec3618475905fa55fece87e/examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java#L46-L48

Reply:
> E.g. if we make `MetricProducerCustomizer.getExporterName` it will be confusing to me: "why customizer of MetricProducers is connected to an exporter"?

This is maybe a reason to rename `MetricProducerCustomizer`. The question is what to rename it to?

Reply:
Some options:
* MetricConsumer
* MetricCollector
* MetricEndpoint

Reply:
Maybe we could unify these interfaces and just simply add MetricProducer to the current interface? If null is returned it would not install the exporter to the SDK.


```java
public interface MetricExporterFactory {
  MetricExporter fromConfig(MetricProducer, Properties);
}
```

Reply:
there's some benefit to having different factories for push and pull cases, since we can support multiple push factories, but only a single pull factory (and not both): https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1281#issuecomment-700834529

Reply:
@pavolloffay @iNikem @anuraaga I think we just need a name for `MetricProducerCustomizer` and then good to merge. 

What do you think of:

```
public interface MetricExporterInitializer {
  void start(MetricProducer, Properties);
}
```

Though maybe that's too close to `MetricExporterFactory`, and people might use `MetricExporterInitializer` accidentally when it would be better to use `MetricExporterFactory` (which can support multiple exporters, and can be configured using `IntervalMetricReader`).

Another option is to go all-in on the one use case we have so far (exposing a metric server):

```
public interface MetricServer {
  void start(MetricProducer, Properties);
}
```

I'm struggling to come up with a name that's both general and not confusing, so would probably opt for the narrower and less confusing `MetricServer`. (credit to @iNikem for this name btw)

Thoughts?